lineapy/plugins/pipeline_writers.py

```diff
@@ -172,44 +176,79 @@ def _write_dag(self) -> None:
 
     def _write_operator_per_session(self) -> str:
         """
-        This hidden method implements Airflow DAG code generation
-        corresponding to the `PythonOperatorPerSession` flavor,
-        where each entire session gets one Python operator. For instance,
-        if the two artifacts in our pipeline (e.g., model and prediction)
-        were created in the same session, we would get an Airflow DAG file
-        looking as the following:
+        This hidden method implements Airflow DAG code generation corresponding
+        to the `PythonOperatorPerSession` flavor, where each session gets its
+        own Python operator. For instance, if the two artifacts in our pipeline
+        (e.g., model and prediction) were created in the same session, we would
+        get an Airflow DAG file looking as the following:
 
         .. code-block:: python
-            import iris_module
+        def dag_setup():
+            pickle_folder = pathlib.Path("/tmp").joinpath("g2_z")
+            if not pickle_folder.exists():
+                pickle_folder.mkdir()
 
-            ...
+
+        def dag_teardown():
+            pickle_files = pathlib.Path("/tmp").joinpath("g2_z").glob("*.pickle")
+            for f in pickle_files:
+                f.unlink()
+
+            def task_run_session_including_g2():
+                artifacts = g2_z_module.run_session_including_g2()
+                pickle.dump(artifacts["g2"], open("/tmp/g2_z/artifact_g2.pickle", "wb"))
+                pickle.dump(artifacts["z"], open("/tmp/g2_z/artifact_z.pickle", "wb"))
 
             with DAG(...) as dag:
-                run_session_including_iris_model = PythonOperator(
-                    task_id="run_session_including_iris_model_task",
-                    python_callable=iris_module.run_session_including_iris_model,
+                setup = PythonOperator(
+                    task_id="dag_setup",
+                    python_callable=dag_setup,
+                )
+
+                teardown = PythonOperator(
+                    task_id="dag_teardown",
+                    python_callable=dag_teardown,
                 )
 
-        This makes the Airflow DAG file more compact but lack transparency
-        and fuller control over each artifact-level task. Check the
-        `PythonOperatorPerArtifact` flavor to address this limitation.
+                run_session_including_g2 = PythonOperator(
+                    task_id="run_session_including_g2_task",
+                    python_callable=task_run_session_including_g2,
+                )
+
+                setup >> run_session_including_g2
+                run_session_including_g2 >> teardown
+
+            run_session_including_g2
```

Comment:
For brevity, I suggest the following "slicing" to highlight parts that are unique to `PythonOperatorPerSession` (NOTE: please ensure indentation is correct):

```python
        .. code-block:: python
            import pickle
            import iris_module

            ...

            def task_run_session_including_g2():
                artifacts = g2_z_module.run_session_including_g2()
                pickle.dump(artifacts["g2"], open("/tmp/g2_z/artifact_g2.pickle", "wb"))
                pickle.dump(artifacts["z"], open("/tmp/g2_z/artifact_z.pickle", "wb"))

            with DAG(...) as dag:
                ...

                run_session_including_g2 = PythonOperator(
                    task_id="run_session_including_g2_task",
                    python_callable=task_run_session_including_g2,
                )

            ...

            run_session_including_g2
```